src/plugins/SetIntegration/set_integration.ts

```diff
@@ -0,0 +1,41 @@
+import { Application } from "probot";
+import { REPO_CORE } from "../../const";
+import { IssueContext } from "../../types";
+import { filterEventNoBot } from "../../util/filter_event_no_bot";
+import { filterEventByRepo } from "../../util/filter_event_repo";
+import { extractIntegrationDocumentationLinks } from "../../util/text_parser";
+
+export const NAME = "SetIntegration";
+
+export const initSetIntegration = (app: Application) => {
+  app.on(
+    ["issues.opened"],
+    filterEventNoBot(
+      NAME,
+      filterEventByRepo(NAME, REPO_CORE, runSetIntegration)
+    )
+  );
+};
+
+export const runSetIntegration = async (context: IssueContext) => {
+  const currentLabels = (await context.github.issues.listLabelsForRepo(
```

Comment:
Should we use a caching function, like we do for PRs? <https://github.com/home-assistant/probot-home-assistant/blob/master/src/util/pull_request.ts#L13>